packages/schematics/angular/migrations/update-6/index.ts

```diff
@@ -542,51 +554,90 @@ function updateSpecTsConfig(config: CliConfig): Rule {
   return (host: Tree, context: SchematicContext) => {
     const apps = config.apps || [];
     apps.forEach((app: AppConfig, idx: number) => {
-      const tsSpecConfigPath =
-        join(app.root as Path, app.testTsconfig || defaults.testTsConfig);
+      const testTsConfig = app.testTsconfig || defaults.testTsConfig;
+      const tsSpecConfigPath = join(normalize(app.root || ''), testTsConfig);
       const buffer = host.read(tsSpecConfigPath);
+
       if (!buffer) {
         return;
       }
-      const tsCfg = JSON.parse(buffer.toString());
-      if (!tsCfg.files) {
-        tsCfg.files = [];
+
+
+      const tsCfgAst = parseJsonAst(buffer.toString(), JsonParseMode.Loose);
+      if (tsCfgAst.kind != 'object') {
+        throw new SchematicsException('Invalid tsconfig. Was expecting an object');
       }
 
-      // Ensure the spec tsconfig contains the polyfills file
-      if (tsCfg.files.indexOf(app.polyfills || defaults.polyfills) === -1) {
-        tsCfg.files.push(app.polyfills || defaults.polyfills);
-        host.overwrite(tsSpecConfigPath, JSON.stringify(tsCfg, null, 2));
+      const filesAstNode = findPropertyInAstObject(tsCfgAst, 'files');
+      if (filesAstNode && filesAstNode.kind != 'array') {
+        throw new SchematicsException('Invalid tsconfig "files" property; expected an array.');
       }
+
+      const recorder = host.beginUpdate(tsSpecConfigPath);
+
+      const polyfills = app.polyfills || defaults.polyfills;
+      if (!filesAstNode) {
+        appendPropertyInAstObject(recorder, tsCfgAst, 'files', [polyfills]);
```

Comment:
If files does not exist, it shouldn't be created. Creating a files array will implicitly exclude all files that are not in the array.

Reply:
Done.